config/application.rb

```diff
@@ -117,18 +117,18 @@ class Application < Rails::Application
           next if source == IdentityConfig.store.domain_name
 
           redirect_uris = Rails.cache.fetch(
-            'all_service_provider_redirect_uris',
+            'all_service_provider_redirect_uris_cors',
             expires_in: IdentityConfig.store.all_redirect_uris_cache_duration_minutes.minutes,
           ) do
-            ServiceProvider.pluck(:redirect_uris).flatten.compact
+            ServiceProvider.pluck(:redirect_uris).flatten.compact.map do |uri|
+              split_uri = uri.split('//')
+              protocol = split_uri[0]
+              domain = split_uri[1].split('/')[0] if split_uri.size > 1
```

Comment:
I know it didn't change in this PR, but as long as we're here, thinking maybe we could consider naming these parts for future us or potentially limiting the split.

```suggestion
              protocol, domain_path = uri.split('//', 2)
              domain, path = domain_path&.split('/')
```

If we're open to it I might try to make this in a follow up PR

Reply:
I'm also considering creating a `lib/identity_cors.rb` or something as a follow up to this and #7786 to pull all of the CORS logic into a single place that the configuration can call.

Reply:
I'm a fan of that!